File: web/src/app/models/appointment.ts
import {Office} from './office';
import {Patient} from './patient';

export class Appointment {
  id: string | number;
  office: Office;
  patient: Patient;
  patientName = '';
  appointmentBegin = '';
  appointmentEnd = '';

  constructor() {
    this.patient = new Patient();
  }

  set _appointmentBegin(appointmentBegin: string) {
    this.appointmentBegin = appointmentBegin.replace('T', ' ' );
  }

  get _appointmentBegin() {
    return this.appointmentBegin.replace(' ', 'T');
  }

  set _appointmentEnd(appointmentEnd: string) {
    this.appointmentEnd = appointmentEnd.replace('T', ' ' );
  }

  get _appointmentEnd() {
    return this.appointmentEnd.replace(' ', 'T');
  }
}

File: mobile/app/src/main/java/wse18/ase/qse03/mobile/model/Appointment.kt
package wse18.ase.qse03.mobile.model

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.io.Serializable
import java.util.*

@JsonIgnoreProperties(ignoreUnknown = true)
data class Appointment(
    var id: Long?,
    var office: Office,
    var patient: User?,
    var patientName: String?,
    var appointmentBegin: Date,
    var appointmentEnd: Date
): Serializable {
    constructor() : this(-1, Office(), User(), "", Date(), Date())
}